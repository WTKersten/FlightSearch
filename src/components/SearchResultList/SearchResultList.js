import React, { PureComponent } from "react";
import PropTypes from "prop-types";

import SearchResultRow from "./SearchResultRow/SearchResultRow";
import "./SearchResultList.scss";

class SearchResultList extends PureComponent {
  render() {
    return (
      <div className="component-search-result-list">
        {this.props.flights.map(flight => (
          <SearchResultRow
            key={flight.id}
            from={flight.from}
            to={flight.to}
            price={flight.price}
          />
        ))}
      </div>
    );
  }
}

SearchResultList.propTypes = {
  flights: PropTypes.array
};

export default SearchResultList;
