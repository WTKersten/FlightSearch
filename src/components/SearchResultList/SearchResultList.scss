
.component-search-result-list {
  margin-top: 40px;
}