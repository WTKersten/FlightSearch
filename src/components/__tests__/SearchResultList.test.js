import React from 'react';
import { shallow } from 'enzyme';
import SearchResultList from '../SearchResultList/SearchResultList';

describe('SearchResultList', () => {
  it('renders correctly', () => {
    const props = [{
      id: 1,
      from: 'Amsterdam (AMS)',
      to: 'London (STD)',
      price: 'â‚¬ 30,00'
    }];
    const component = shallow(<SearchResultList flights={props} />);
    expect(component).toMatchSnapshot();
  });
});