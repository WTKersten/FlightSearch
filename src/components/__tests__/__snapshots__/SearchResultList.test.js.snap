// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultList renders correctly 1`] = `
<div
  className="component-search-result-list"
>
  <SearchResultRow
    from="Amsterdam (AMS)"
    key="1"
    price="â‚¬ 30,00"
    to="London (STD)"
  />
</div>
`;
